### ISSUE TITLE
Add SUBSCRIBE command support

### ISSUE THREAD

**Issue by truedat101** [adilbaig/Tiny-Redis#15]
http://redis.io/commands/subscribe

You mentioned in the README this just needs a little work.  I've added a hack-ish implementation in my fork.  I will test for a bit before submitting any kind of pull request.  Your code is nice and clean and mine is ...

**Comment by adilbaig** [comment]
Hey,

Thanks for putting your time into this. This would be a great addition. I happened to peek into your fork, just to get a feel of the work. Good progress! 

I'd like to share a quick thought on this. I intend to keep this library thread free. Any introduction of threads means it will require careful consideration on the user's part to use it, especially as a lower level abstraction for higher tools. I don't want to limit the audience this way. 

My idea for SUBSCRIBE would be to create a separate Subscribe object with a 'socket.accept' loop, and just return Responses as they come in. That object can only unsubscribe. If a user intends to SUBSCRIBE and issue Redis commands they'll need to open two socket connections and deal with Threading themselves.

Thoughts?

**Comment by truedat101** [comment]
I agree on the "no threads" approach.  It pushes the baggage up a level to the application developer.

I took a survey of other client libs out there and it seems they range the expanse from super simple (incomplete) all the way to over the top with async, sync, etc... and some require an external event lib.  To keep with the spirit of Redis I think it makes sense to keep it lightweight and simple with few dependencies and easy inspection of how it works.

**Comment by SommerEngineering** [comment]
Any news about this implementation? It would be a great extension to have PUB/SUB support e.g. to implement microservices, etc. Keep the good work :+1:

**Comment by truedat101** [comment]
I made something (not completely in the spirit of single threaded design desired by the author).  At the end of the day, I found the author maybe was on to something.  Due to thread-local constraints, sharing data between threads is a problem in the current design.  I thought originally I could use Fibers or Threads, but ran into numerous design issues trying this route.  I think the above comment from the author suggests the right approach .  Others who know D better than I may suggest how to get an alternative working.
